x-pack/plugins/security_solution/common/endpoint/types/actions.ts

```diff
@@ -9,36 +9,45 @@ import { TypeOf } from '@kbn/config-schema';
 import { HostIsolationRequestSchema } from '../schema/actions';
 
 export type ISOLATION_ACTIONS = 'isolate' | 'unisolate';
-
 export interface EndpointAction {
-  action_id: string;
-  '@timestamp': string;
-  expiration: string;
-  type: 'INPUT_ACTION';
-  input_type: 'endpoint';
-  agents: string[];
-  user_id: string;
-  data: {
-    command: ISOLATION_ACTIONS;
-    comment?: string;
+  type: 'action';
+  item: {
+    action_id: string;
+    '@timestamp': string;
+    expiration: string;
+    type: 'INPUT_ACTION';
+    input_type: 'endpoint';
+    agents: string[];
+    user_id: string;
+    data: {
+      command: ISOLATION_ACTIONS;
+      comment?: string;
+    };
   };
 }
 
 export interface EndpointActionResponse {
-  '@timestamp': string;
-  /** The id of the action for which this response is associated with */
-  action_id: string;
-  /** The agent id that sent this action response */
-  agent_id: string;
-  started_at: string;
-  completed_at: string;
-  error: string;
-  action_data: {
-    command: ISOLATION_ACTIONS;
-    comment?: string;
+  type: 'response';
+  item: {
+    '@timestamp': string;
+    /** The id of the action for which this response is associated with */
+    action_id: string;
+    /** The agent id that sent this action response */
+    agent_id: string;
+    started_at: string;
+    completed_at: string;
+    error: string;
```

Comment:
So here, I feel that `error` key should be either empty ('') or `undefined` when the isolate/unisolate action was successful. 

Alternatively, it could be that either an `error` or a `success` key is present in the response data and not both, indicating that the action was successful or not. Not sure if that's how the index works.

Reply:
> So here, I feel that `error` key should be either empty ('') or `undefined` when the isolate/unisolate action was successful.

When it is successful, I agree. `error` will either be blank or omitted. Note that that may _also_ be the case if unsuccessful. This error might only be populated when the failure generates from "above" endpoint in the call chain.

Reply:
@pzl thanks for clarifying. So when a failure occurs doesn't it mean that the action has failed? Or does it self-correct and carry on? 🤔 

As I understand, looking at the `error` key now won't tell us if it is a successful or failed action. What can we do so we can distinguish between the two?